src/App.js

```diff
@@ -1,26 +1,80 @@
 import React from 'react';
-import logo from './logo.svg';
 import './App.css';
+import 'bulma/css/bulma.css';
+import FoodBox from './components/FoodBox'
+import foods from './foods.json';
+import AddFood from './components/AddFood'
+import TodaysFoods from './components/TodaysFoods'
+import FoodForm from './components/FoodForm'
+import searchBar from './components/searchBar'
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+
+  constructor(props){
+    super(props); // read more on this
+    this.state = {
+    showForm: false,  
+    foods: foods,
+    filtered: foods,
+    TodaysFoods: { },
+    };
+
+}
+
+  handleAddfood = (event) => {
+    event.preventDefault() // always do this with submit events
+    let foodName = event.currentTarget.food.value
+    let newfoods = [{title:foodName, completed: false}, ...this.state.foods]
+    this.setState({
+      foods: newfoods,
+    })   
+  }
+
+  handleQuantityChange = (event) => {
+    const { name, value, quantity } = event.target;
+    this.setState({
+      name: name,
+      value: value,
+      quantity: quantity
+    })   
+  }
+
+  filteredFood = (e) => {
+    const searchText = e.currentTarget.value 
+    const filtered = this.state.foods.filter((food)=> {
+      return food.name.toLowerCase().includes(searchText.toLowerCase())
+  })
+  this.setState({
+    foods: filtered
+  })
+}
+
+showForm = () => {
+  this.setState({
+    showForm: true,
+  })
+}
+
+    render(){
+    return (
+      <div>
+      <div className= 'IronContacts'>
+      <h2>IronNutrition</h2><searchBar filteredFood={this.filteredFood} />
+      
+      <FoodForm handleChange={this.filteredFood} name='search' type='text' />
```

Comment:
you have two search bars. The one above that was not working because of how it was written, and this one that is not doing anything.